src/Authenticator/LoginHandler.php

```diff
@@ -144,9 +149,58 @@ public function getSchema()
      * @param HTTPRequest $request
      * @return HTTPResponse
      */
-    public function startRegister(HTTPRequest $request)
+    public function startRegistration(HTTPRequest $request)
     {
-        return $this->jsonResponse(['errors' => ['Registration not yet implemented']], 500);
+        $sessionStore = $this->getSessionStore();
+        $sessionMember = $sessionStore->getMember();
+        $loggedInMember = Security::getCurrentUser();
+
+        if (is_null($loggedInMember) && is_null($sessionMember)) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.NOT_AUTHENTICATING', 'You must be logged or logging in')]],
+                403
+            );
+        }
+
+        // If the user isn't fully logged in and they already have a registered method, they can't register another
+        if (is_null($loggedInMember) && $sessionMember && count($sessionMember->RegisteredMFAMethods()) > 0) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.MUST_USE_EXISTING_METHOD', 'This member already has an MFA method')]],
+                400
+            );
+        }
+
+        // Fall back to the logged in user at this point if not in the login process
+        $member = $sessionMember ?: $loggedInMember;
+
+        // Pull a method to register from the request
+        $specifiedMethod = str_replace('-', '\\', $request->param('Method'));
+
+        // Sanity check that the method hasn't already been registered
+        $existingRegisteredMethod = $this->getMethodRegistry()->getMethodFromMember($member, $specifiedMethod);
+
+        if ($existingRegisteredMethod) {
+            return $this->jsonResponse(
+                [
+                    'errors' => [
+                        _t(
+                            __CLASS__ . '.METHOD_ALREADY_REGISTERED',
+                            'That method has already been registered against this Member'
+                        )
+                    ]
+                ],
+                400
+            );
+        }
+
+        // Mark the given method as started within the session
+        $sessionStore->setMethod($specifiedMethod);
+        // Allow the registration handler to begin the process and generate some data to pass through to the front-end
+        $data = Injector::inst()->create($specifiedMethod)->getRegisterHandler()->start($sessionStore);
```

Comment:
I could pass `SilverStripe\CMS\Model\SiteTree` as `$request->param('Method')` and get a fatal error getRegisterHandler not found - maybe do a typecheck on the interface and throw an error if it doesn't implement it